src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java

```diff
@@ -117,6 +118,12 @@ private void addRoleIfNotPresent(User user, RoleEnum roleName) {
         }
     }
 
+    private void deleteDepartmentIfPresentAndUserIsNotAdminOrSuperAdmin(User user) {
+        if (user.getDepartment() != null && user.getRoles().stream().noneMatch(role -> role.getName().equals(RoleEnum.ADMIN) || role.getName().equals(RoleEnum.SUPER_ADMIN))) {
```

Comment:
this `user.getRoles().stream().noneMatch(role -> role.getName().equals(RoleEnum.ADMIN) || role.getName().equals(RoleEnum.SUPER_ADMIN))` feels a bit clunky coming from JS but not sure how else you'd do it in Java, so happy with it as-is

Reply:
Only other way I can think is by extracting the 2 predicates and do an `or` comparison